[move] Shared objects must be newly created

tnowacki opened the pull request:
- Throw abort on sui::transfer::shared if the object is not new
- This means it cannot be previously owned or wrapped
- Fixes #5835

- We might also need to update docs/examples?

sblackshear commented on crates/sui-framework/src/natives/object_runtime/mod.rs:
```diff
@@ -123,12 +129,25 @@ impl<'a> ObjectRuntime<'a> {
             .value_as::<AccountAddress>()?
             .into();
+        let transfer_result =
+            if self.state.new_ids.contains_key(&id) || id == SUI_SYSTEM_STATE_OBJECT_ID {
```
Second part of this conditional is kind of surprising--is this a special case for genesis? Might be worth mentioning if so.

lxfind replied:
Yeah worth a comment. The system state object is special in that its ID is not generated from TxContext, but hard coded at 0x5. It maybe clearer if we could find a cleaner way to augment new_ids to include it in the genesis case.

gdanezis commented:
Thanks for the simplification (tag: @lxfind )

mystenmark reviewed (commented):
Thanks for doing this!

lxfind commented:
Just double check, it can detect sharing unwrapped objects as well, right?

tnowacki commented:
> Just double check, it can detect sharing unwrapped objects as well, right?

Correct, I added a block of comments in there for that case too.

tnowacki commented:
> > Just double check, it can detect sharing unwrapped objects as well, right?
> 
> Correct, I added a block of comments in there for that case too.

I updated the PR title and description to make this a bit more clear
